commands/utils/quick-pick-azure.ts

```diff
@@ -33,6 +34,23 @@ export async function quickPickACRRepository(registry: Registry, prompt?: string
     return desiredRepo.data;
 }
 
+export async function quickPickBuildTask(registry: Registry, subscription: Subscription, resourceGroup: ResourceGroup, prompt?: string): Promise<ContainerModels.BuildTask> {
+    const placeHolder = prompt ? prompt : 'Choose a Build Task';
+    const client = AzureUtilityManager.getInstance().getContainerRegistryManagementClient(subscription);
+
+    let buildTasks: ContainerModels.BuildTask[] = [];
+    let buildTaskNames: string[] = [];
+    buildTasks = await client.buildTasks.list(resourceGroup.name, registry.name);
+
+    for (let bt of buildTasks) {
+        buildTaskNames.push(bt.name);
+    }
+    let desiredBuildTask = await vscode.window.showQuickPick(buildTaskNames, { 'canPickMany': false, 'placeHolder': placeHolder });
```

Comment:
use ext.ui.showQuickPick, see the others to have a basis on how to do it. This does consistent error handling and has the added benefit of having the recently selected thingy if you wrap items in the azurequickpick item type.